### ISSUE TITLE
Can't build on macOS 10.13.4

### ISSUE THREAD

**Issue by hpoit** [JuliaIO/HDF5.jl#482]
I would like to help, but now trying to understand the problem(s)
```
julia> using HDF5
INFO: Precompiling module HDF5.
ERROR: LoadError: HDF5 not properly installed. Please run Pkg.build("HDF5")
Stacktrace:
 [1] error(::String) at ./error.jl:21
 [2] include_from_node1(::String) at ./loading.jl:576
 [3] include(::String) at ./sysimg.jl:14
 [4] anonymous at ./<missing>:2
while loading /Users/kevinliu/.julia/v0.6/HDF5/src/HDF5.jl, in expression starting on line 38
ERROR: Failed to precompile HDF5 to /Users/kevinliu/.julia/lib/v0.6/HDF5.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:710
 [2] _require(::Symbol) at ./loading.jl:497
 [3] require(::Symbol) at ./loading.jl:405

julia> Pkg.build("HDF5")
INFO: Building Homebrew
Updated 1 tap (homebrew/core).
==> Updated Formulae
annie                      fio                        storm
ansible                    fn                         teleport
aws-sdk-cpp                frugal                     youtube-dl
bash                       jenkins-job-builder
django-completion          proj
INFO: Building CMakeWrapper
INFO: Building Blosc
INFO: Building HDF5
==> Installing gcc from staticfloat/juliatranslated
==> Downloading https://homebrew.bintray.com/bottles/gcc-8.1.0.high_sierra.bottl
Already downloaded: /Users/kevinliu/Library/Caches/Homebrew.jl/gcc-8.1.0.high_sierra.bottle.1.tar.gz
==> Pouring gcc-8.1.0.high_sierra.bottle.1.tar.gz
Error: Failed changing install name in /Users/kevinliu/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libubsan.1.dylib
  from @@HOMEBREW_CELLAR@@/gcc/8.1.0/lib/gcc/8/i386/libstdc++.6.dylib
    to /Users/kevinliu/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libstdc++.6.dylib
Error: Updated load commands do not fit in the header of /Users/kevinliu/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libubsan.1.dylib. /Users/kevinliu/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libubsan.1.dylib needs to be relinked, possibly with -headerpad or -headerpad_max_install_names
Warning: Bottle installation failed: building from source.
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-8.1.0/gcc-8.1.0.tar.xz
Already downloaded: /Users/kevinliu/Library/Caches/Homebrew.jl/gcc-8.1.0.tar.xz
==> ../configure --build=x86_64-apple-darwin17.5.0 --prefix=/Users/kevinliu/.jul
==> make
^CERROR: InterruptException:
Stacktrace:
 [1] process_events at ./libuv.jl:82 [inlined]
 [2] wait() at ./event.jl:216
 [3] wait(::Condition) at ./event.jl:27
 [4] stream_wait(::Base.Process, ::Condition, ::Vararg{Condition,N} where N) at ./stream.jl:42
 [5] wait at ./process.jl:799 [inlined]
 [6] success(::Base.Process) at ./process.jl:665
 [7] success(::Base.CmdRedirect) at ./process.jl:678
 [8] build(::String, ::String, ::String) at ./pkg/entry.jl:616
 [9] build!(::Array{String,1}, ::Set{Any}, ::String) at ./pkg/entry.jl:627
 [10] build!(::Array{String,1}, ::Dict{Any,Any}, ::Set{Any}) at ./pkg/entry.jl:635
 [11] build(::Array{String,1}) at ./pkg/entry.jl:650
 [12] ===============================================================================[ ERROR: HDF5 ]================================================================================

LoadError: InterruptException:
while loading /Users/kevinliu/.julia/v0.6/HDF5/deps/build.jl, in expression starting on line 41

==============================================================================================================================================================================
(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}})() at ./pkg/dir.jl:36
 [13] cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}}, ::String) at ./file.jl:70
 [14] #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at ./pkg/dir.jl:36
 [15] build(::String, ::Vararg{String,N} where N) at ./pkg/pkg.jl:254

julia> 
```

**Comment by musm** [comment]
Hmm I have no expertise on mac, the error seems strange but related to Homebrew. I hope someone else can provide some help here. Maybe also post this under Homebrew.jl

**Comment by stillyslalom** [comment]
See https://github.com/JuliaIO/HDF5.jl/issues/483#issuecomment-395146413

(just let the gcc build run to completion, although it may take several hours)

**Comment by nickeubank** [comment]
Same on 10.13.5. 

@stillyslalom I presume that works (not only does it take hours, but it takes a ton of CPU attention... can't work while running, so had to abort), but is there a better fix in the works?

**Comment by stillyslalom** [comment]
@nickeubank: not a fix, but a workaround https://github.com/JuliaIO/HDF5.jl/issues/483#issuecomment-403721993

**Comment by julien-vaes** [comment]
I have the same issue on 10.13.5

**Comment by stillyslalom** [comment]
@julien-vaes does the linked workaround work for you? The underlying issue is upstream and might not be fixed for some time.

**Comment by julien-vaes** [comment]
@stillyslalom thank your for your help! I am still waiting to see if it works, for the moment I have already some errors and it is evolving anymore:

`julia> using Homebrew

julia> raw = "https://raw.githubusercontent.com/Homebrew/homebrew-core/858282eb2f1381e027e3db225da11fd2dbb13c4d/Formula/gcc.rb"
"https://raw.githubusercontent.com/Homebrew/homebrew-core/858282eb2f1381e027e3db225da11fd2dbb13c4d/Formula/gcc.rb"

julia> Homebrew.brew(`install --force-bottle $raw`)

######################################################################## 100.0%
Warning: gcc 8.1.0 is available and more recent than version 7.3.0.
==> Downloading https://homebrew.bintray.com/bottles/gcc-7.3.0.high_sierra.bottle.1.tar.gz
==> Downloading from https://akamai.bintray.com/d0/d09669e3679bb54448f00cda4bf520e631f7487f132ebfe6e0d2f6ec
######################################################################## 100.0%
==> Pouring gcc-7.3.0.high_sierra.bottle.1.tar.gz
🍺  /Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/7.3.0: 1,486 files, 284.9MB

julia> Homebrew.brew(`switch gcc 7.3.0`)
Cleaning /Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/7.3.0
33 links created for /Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/7.3.0

julia> Homebrew.brew(`pin gcc`)

julia> Homebrew.update()
Already up-to-date.
Uninstalling gcc... (1,486 files, 284.9MB)
==> Installing gcc from staticfloat/juliatranslated
==> Downloading https://homebrew.bintray.com/bottles/gcc-8.1.0.high_sierra.bottle.1.tar.gz
Already downloaded: /Users/jvaes/Library/Caches/Homebrew.jl/gcc-8.1.0.high_sierra.bottle.1.tar.gz
==> Pouring gcc-8.1.0.high_sierra.bottle.1.tar.gz
Error: Failed changing install name in /Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libubsan.1.dylib
  from @@HOMEBREW_CELLAR@@/gcc/8.1.0/lib/gcc/8/i386/libstdc++.6.dylib
    to /Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libstdc++.6.dylib
Error: Updated load commands do not fit in the header of /Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libubsan.1.dylib. /Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cellar/gcc/8.1.0/lib/gcc/8/i386/libubsan.1.dylib needs to be relinked, possibly with -headerpad or -headerpad_max_install_names
Warning: Bottle installation failed: building from source.
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-8.1.0/gcc-8.1.0.tar.xz
Already downloaded: /Users/jvaes/Library/Caches/Homebrew.jl/gcc-8.1.0.tar.xz
==> ../configure --build=x86_64-apple-darwin17.6.0 --prefix=/Users/jvaes/.julia/v0.6/Homebrew/deps/usr/Cell
==> make`

**Comment by musm** [comment]
stale, I think also fixed by using BinaryBuilder
